test/text.c

```diff
@@ -0,0 +1,358 @@
+{
+    u32 i;
+    // -6 for the question mark in FONT_NORMAL.
+    const u32 fontId = FONT_NARROWER, widthPx = 102 - 6;
+    u32 item = ITEM_NONE;
+    u8 pluralName[ITEM_NAME_PLURAL_LENGTH + 1];
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    CopyItemNameHandlePlural(item, pluralName, 2);
+    EXPECT_LE(GetStringWidth(fontId, pluralName, 0), widthPx);
+}
+
+TEST("Item plural names fit on PC storage (left box)")
+{
+    u32 i;
+    // -6 for the question mark in FONT_NORMAL.
+    const u32 fontId = FONT_NARROWER, widthPx = 13 * 8 - 6;
+    u32 item = ITEM_NONE;
+    u8 pluralName[ITEM_NAME_PLURAL_LENGTH + 1];
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    CopyItemNameHandlePlural(item, pluralName, 2);
+    EXPECT_LE(GetStringWidth(fontId, pluralName, 0), widthPx);
+}
+
+TEST("Item names fit on Pokemon Storage System")
+{
+    u32 i;
+    const u32 fontId = FONT_SMALL_NARROWER, widthPx = 7 * 8 - 6;
+    u32 item = ITEM_NONE;
+    for (i = 1; i < ITEMS_COUNT; i++)
+    {
+        if (gItemsInfo[i].importance) continue;
+        PARAMETRIZE_LABEL("%S", gItemsInfo[i].name) { item = i; }
+    }
+    switch (item)
+    {
+    case ITEM_ENERGY_POWDER:
+    case ITEM_PEWTER_CRUNCHIES:
+    case ITEM_RAGE_CANDY_BAR:
+    case ITEM_LUMIOSE_GALETTE:
+    case ITEM_HEALTH_FEATHER:
+    case ITEM_MUSCLE_FEATHER:
+    case ITEM_RESIST_FEATHER:
+    case ITEM_GENIUS_FEATHER:
+    case ITEM_CLEVER_FEATHER:
+    case ITEM_ABILITY_CAPSULE:
+    case ITEM_DYNAMAX_CANDY:
+    case ITEM_MAX_MUSHROOMS:
+    case ITEM_GOLD_BOTTLE_CAP:
+    case ITEM_PRETTY_FEATHER:
+    case ITEM_STRANGE_SOUVENIR:
+    case ITEM_FOSSILIZED_BIRD:
+    case ITEM_FOSSILIZED_FISH:
+    case ITEM_FOSSILIZED_DRAKE:
+    case ITEM_FOSSILIZED_DINO:
+    case ITEM_SURPRISE_MULCH:
+    case ITEM_YELLOW_APRICORN:
+    case ITEM_GREEN_APRICORN:
+    case ITEM_WHITE_APRICORN:
+    case ITEM_BLACK_APRICORN:
+    case ITEM_THUNDER_STONE:
+    case ITEM_GALARICA_WREATH:
+    case ITEM_STRAWBERRY_SWEET:
+    case ITEM_ELECTRIC_MEMORY:  // *
+    case ITEM_FIGHTING_MEMORY:  // *
+    case ITEM_GROUND_MEMORY:    // *
+    case ITEM_PSYCHIC_MEMORY:   // *
+    case ITEM_DRAGON_MEMORY:    // *
+    case ITEM_CHARIZARDITE_X:   // *
+    case ITEM_CHARIZARDITE_Y:   // *
+    case ITEM_ULTRANECROZIUM_Z: // *
+    case ITEM_DEEP_SEA_SCALE:   // *
+    case ITEM_DEEP_SEA_TOOTH:   // *
+    case ITEM_NEVER_MELT_ICE:   // *
+    case ITEM_WEAKNESS_POLICY:  // *
+    case ITEM_SAFETY_GOGGLES:   // *
+    case ITEM_ADRENALINE_ORB:   // *
+    case ITEM_TERRAIN_EXTENDER: // *
+    case ITEM_PROTECTIVE_PADS:  // *
+    case ITEM_HEAVY_DUTY_BOOTS: // *
+    case ITEM_UTILITY_UMBRELLA: // *
+    case ITEM_MARANGA_BERRY:    // *
+    case ITEM_PUNCHING_GLOVE:   // *
+    case ITEM_AUSPICIOUS_ARMOR:
+    case ITEM_BOOSTER_ENERGY:   // *
```

Comment:
What are these about?

Reply:
The stars are next to items that have hold effects (i.e. are likely to be held by a Pokémon). Doesn't really do anything, except if somebody wanted to address just those ones.

Reply:
I'm not sure it's relevant to this test. Even if they have no use when held, items would still be displayed in UIs and such.

Reply:
Yeah, that's true.

I think there's some use in being able to say "these are the important items, so I'm going to change the UI so that at least these ones fit, but I don't mind if some of the items which don't have hold effects don't quite fit".

But I can take the `*`s away if we don't think it's useful :)

Reply:
Maybe they could be grouped like this then
```c
    case ITEM_BLACK_APRICORN:
    case ITEM_THUNDER_STONE:
    case ITEM_GALARICA_WREATH:
    case ITEM_STRAWBERRY_SWEET:
    // Items with hold effects.
    case ITEM_ELECTRIC_MEMORY:
    case ITEM_FIGHTING_MEMORY:
    case ITEM_GROUND_MEMORY:
    case ITEM_PSYCHIC_MEMORY:
```

Reply:
Oh, that's way better than what I wrote!